test/robosherlock_utest/test_storage.cpp

```diff
@@ -0,0 +1,269 @@
+
+  cas = engine->newCAS();
+
+  if(cas == NULL)
+  {
+    std::cerr<<"Creating new CAS failed."<<std::endl;
+    engine->destroy();
+    delete engine;
+    return 0;
+  }
+
+  UnicodeString ustrInputText;
+  ustrInputText.fromUTF8(name);
+  cas->setDocumentText(uima::UnicodeStringRef(ustrInputText));
+  std::cerr<<"processing CAS"<<std::endl;
+ 
+  const uima::AnalysisEngineMetaData &aeMetaData = engine->getAnalysisEngineMetaData();
+  std::string aeDescription;
+  aeMetaData.getDescription().toUTF8String(aeDescription);
+  uima::AnnotatorContext &annotContext = engine->getAnnotatorContext();
+  uima::AnnotatorContext::TyMapDelegateAnCs delegates =  annotContext.getDelegates();
+  	
+  //prints all annotatores imported in the xml (not just in the fixed flow)
+  std::cerr<<"========================================"<<std::endl;
+  for(auto d:delegates)
+  {
+      std::string out;
+      d.first.toUTF8String(out);
+      std::cerr<<out<<std::endl;
+  }
+  std::cerr<<"========================================"<<std::endl;
+
+ try
+    {
+      //we process here
+      uima::CASIterator casIter = engine->processAndOutputNewCASes(*cas);
+      rs::SceneCas sceneCas(*cas);
+      cv::Mat colorImg;
+      sceneCas.get(VIEW_COLOR_IMAGE,colorImg);
+      initial_image_width = colorImg.size().width;
+      initial_image_height = colorImg.size().height;
+      engine->getAnnotatorContext().releaseCAS(*cas);
+    }
+    catch(const rs::FrameFilterException &)
+    {
+      outError("There is an error in the rs");
+    }
+     //cout << "\n \n \n \n**************************************************************************** \n \n \n"; cin.get();
+  UnicodeString ucs_delegate("CollectionReader");
+  uima::AnnotatorContext *cr_context =  annotContext.extractDelegate(ucs_delegate);
+
+  if(cr_context->isParameterDefined("camera_config_files"))
+  {
+    std::vector<std::string *> values;
+    cr_context->extractValue("camera_config_files", values);
+    for (auto v:values)
+    {
+       std::cerr<<*v<<std::endl;
+	
+    }
+  }
+
+  std::vector<UnicodeString> new_configs;
+  new_configs.push_back(UnicodeString("config_mongodb_playback_utest.ini"));
+   
+  cr_context->assignValue(UnicodeString("camera_config_files"),new_configs);
+  engine->reconfigure();
+  
+  try
+  {
+  uima::CASIterator casIter = engine->processAndOutputNewCASes(*cas);
+  rs::SceneCas sceneCas(*cas);
+  cv::Mat colorImg;
+  sceneCas.get(VIEW_COLOR_IMAGE,colorImg);
+  db_image_width = colorImg.size().width;
+  db_image_height = colorImg.size().height;
+  //cout <<"New height is "<< db_image_height;
+  engine->getAnnotatorContext().releaseCAS(*cas);
+  }
+  catch(const rs::FrameFilterException &){}
+  engine->collectionProcessComplete();
+  cout << "This is the end of the program\n";
+  cout << "Conclusion db image and initial image equality is: " << (db_image_width == initial_image_width);
+  engine->destroy();
+  testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+
+}
```

Comment:
this looks good so far; remove the help function; it is not needed; Also remove all the extra outputs; they can be extra checks in the Unit test instead;